src/test/java/org/broadinstitute/consent/http/service/EmailServiceTest.java

```diff
@@ -554,4 +557,112 @@ void sendDarExpirationReminderMessage() throws Exception {
+    user2.setUserId(124);
+    user2.setDisplayName("Jane Doe");
+    user2.setEmail("jd@somewhereelse");
+
+    DataAccessRequest dar1 = getMockedDar("DAR-12345", UUID.randomUUID().toString(), user1);
+    DataAccessRequest dar2 = getMockedDar("DAR-12346", UUID.randomUUID().toString(), user2);
+
+    when(userDAO.findUserById(user1.getUserId())).thenReturn(user1);
+    when(userDAO.findUserById(user2.getUserId())).thenReturn(user2);
+    List<DataAccessRequest> dars = List.of(dar1, dar2);
+    when(dataAccessRequestDAO.findAgedDARsByEmailTypeOlderThanInterval(any(), any(), any())).thenReturn(dars);
+    when(templateHelper.getTemplate(EmailType.DAR_EXPIRATION_REMINDER.templateName)).thenReturn(mock());
+    when(templateHelper.getTemplate(EmailType.DAR_EXPIRED.templateName)).thenReturn(mock());
+
+    assertDoesNotThrow(()->service.sendExpirationNotices());
+    verify(emailDAO).insert(eq(dar1.getReferenceId()),eq(null), eq(user1.getUserId()), eq(EmailType.DAR_EXPIRED.getTypeInt()), any(), any(), any(), any(), any());
+    verify(emailDAO).insert(eq(dar1.getReferenceId()),eq(null), eq(user1.getUserId()), eq(EmailType.DAR_EXPIRATION_REMINDER.getTypeInt()), any(), any(), any(), any(), any());
+    verify(emailDAO).insert(eq(dar2.getReferenceId()),eq(null), eq(user2.getUserId()), eq(EmailType.DAR_EXPIRED.getTypeInt()), any(), any(), any(), any(), any());
+    verify(emailDAO).insert(eq(dar2.getReferenceId()),eq(null), eq(user2.getUserId()), eq(EmailType.DAR_EXPIRATION_REMINDER.getTypeInt()), any(), any(), any(), any(), any());
+    verify(emailDAO, times(4)).insert(any(), any(), any(), any(), any(), any(), any(), any(), any());
+  }
+
+  @Test
+  void sendExpirationNoticesTestMissingEmailForOneUser() throws IOException {
+    ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+    ch.qos.logback.classic.Logger log = (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(EmailService.class);
+    listAppender.start();
+    log.addAppender(listAppender);
+    User user1 = new User();
+    user1.setUserId(123);
+    user1.setDisplayName("John Doe");
+    user1.setEmail("jd@somewhere");
+
+    User user2 = new User();
+    user2.setUserId(124);
+    user2.setDisplayName("Jane Doe");
+
+    DataAccessRequest dar1 =getMockedDar("DAR-12345", UUID.randomUUID().toString(), user1);
+    DataAccessRequest dar2 = getMockedDar("DAR-12346", UUID.randomUUID().toString(), user2);
+
+    when(userDAO.findUserById(user1.getUserId())).thenReturn(user1);
+    when(userDAO.findUserById(user2.getUserId())).thenReturn(user2);
+
+    List<DataAccessRequest> dars = List.of(dar2, dar1);
+
+    when(dataAccessRequestDAO.findAgedDARsByEmailTypeOlderThanInterval(any(), any(), any())).thenReturn(dars);
+    when(templateHelper.getTemplate(EmailType.DAR_EXPIRATION_REMINDER.templateName)).thenReturn(mock());
+    when(templateHelper.getTemplate(EmailType.DAR_EXPIRED.templateName)).thenReturn(mock());
+
+    assertDoesNotThrow(()->service.sendExpirationNotices());
```

Comment:
Instead of using a log appender, you can use a mock logger and then you can use `verify()` as is used for other mocked APIs.
```suggestion
    Logger logger = mock();
    try (var logFactoryMock = mockStatic(LoggerFactory.class)) {
      logFactoryMock.when(() -> LoggerFactory.getLogger(EmailService.class)).thenReturn(logger);
      service.sendExpirationNotices();
    }
    verify(logger, times(2)).warn(any());
```

Reply:
Had something like your suggestion and it worked for the test and the class, but broke tests in another class when running the full suite.  Settled on the approach used because the scope of this ticket is too big already.

Reply:
> but broke tests in another class when running the full suite

Interesting, I guess the static nature of the logger initialization means that it will only happen once across a test suite. Maybe using a spy() here would work, but that would be a bit more work and definitely not worth doing right now.